tests/framework/results/test_context.py

```diff
@@ -0,0 +1,86 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from vivarium.framework.results.context import ResultsContext
+from vivarium.framework.results.stratification import Stratification
+
+from .mocks import (
+    CATEGORIES,
+    NAME,
+    SOURCES,
+    sorting_hat_bad_mapping,
+    sorting_hat_serial,
+    sorting_hat_vector,
+)
+
+
+@pytest.mark.parametrize(
+    "name, sources, categories, mapper, is_vectorized",
+    [
+        (  # expected Stratification for vectorized
+            NAME,
+            SOURCES,
+            CATEGORIES,
+            sorting_hat_vector,
+            True,
+        ),
+        (  # expected Stratification for non-vectorized
+            NAME,
+            SOURCES,
+            CATEGORIES,
+            sorting_hat_serial,
+            False,
+        ),
+    ],
+)
+def test_add_stratification(name, sources, categories, mapper, is_vectorized):
+    ctx = ResultsContext()
+    ctx.add_stratification(name, sources, categories, mapper, is_vectorized)
+    expected_object = Stratification(name, sources, categories, mapper, is_vectorized)
+    assert expected_object in ctx._stratifications
+
+
+@pytest.mark.parametrize(
+    "name, sources, categories, mapper, is_vectorized, expected_exception",
+    [
+        (  # map to a category that isn't in CATEGORIES
```

Comment:
I think each of these cases should be a separate test, since they have nothing in common other than the fact that they should fail

Reply:
I removed these tests, as they were nonsensical and replaced it with the only real exception testing I could do for this bit.